Marketplace/src/main/java/com/example/demo/config/DataSourceConfig.java

```diff
@@ -3,28 +3,49 @@
 import javax.sql.DataSource;
 
 import org.flywaydb.core.Flyway;
-import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
 import org.springframework.boot.jdbc.DataSourceBuilder;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
 
 @Configuration
 public class DataSourceConfig {
 
 	@Bean
-	public DataSource dataSource(@Value("${spring.datasource.driverClassName}") String driverClassName, 
-								 @Value("${spring.datasource.url}") String url,
-								 @Value("${spring.datasource.username}") String username,
-								 @Value("${spring.datasource.password}") String password) {
+	public Path imgDirectory() {
+		return Paths.get("img");
+	}
+
+	@Bean
+	public JdbcTemplate jdbcTemplate(DataSource dataSource) {
+		return new JdbcTemplate(dataSource);
+	}
+
+	@Bean DbProperties dbProperties() {
```

Comment:
Remove this bean. You already have DbProperties bean with @ConfigurationProperties.  Simply inject it to dataSource() method